lib/publiccloud/provider.pm

```diff
@@ -359,7 +366,12 @@ sub terraform_apply {
 
     my $cmd = 'terraform plan -no-color ';
     if (!get_var('PUBLIC_CLOUD_SLES4SAP')) {
-        $cmd .= "-var 'image_id=" . $image . "' ";
+        # Some auxiliary variables, requires for fine control and public cloud provider specifics
+        my $offer = get_var("PUBLIC_CLOUD_AZURE_OFFER");
+        my $sku   = get_var("PUBLIC_CLOUD_AZURE_SKU");
+        $cmd .= "-var 'image_id=" . $image . "' " if ($image);
+        $cmd .= "-var 'offer=" . $offer . "' "    if (is_azure && $offer);
+        $cmd .= "-var 'sku=" . $sku . "' "        if (is_azure && $sku);
```

Comment:
What do you think of this idea: https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/0a640e2f10f6c4217fb8af7e6e1b5c24f08ce544

For me it is more usable for future provider specific variables ...

Reply:
:+1: :+1: :+1: 

Adapting it right now.